torch/nn/utils/_stateless.py

```diff
@@ -0,0 +1,56 @@
+import contextlib
+
+import torch
+
+
+def _get_module_parameters_and_buffers(module):
+    """Helper function that returns a dictionary
+    with only parameters and buffers, we don't use state_dict as it might
+    be polluted with module attributes.
+    """
+    parameters_and_buffers = dict(module.named_parameters())
+    parameters_and_buffers.update(dict(module.named_buffers))
+    # TODO: clean entries that are already parameterized?
+    # params with weight norm/spectral norm applied
+    return parameters_and_buffers
+
+@contextlib.contextmanager
+def reparametrize_module(module, parameters_and_buffers):
+    # Parametrization does not support to change submodules directly
+    for name, tensor in parameters_and_buffers.items():
+        _reparametrize_in_submodule(module, name.split("."), tensor)
+    yield
+    for name in parameters_and_buffers:
+        _remove_reparametrize_in_submodule(module, name.split("."))
+
+class _ReparametrizedTensor(torch.nn.Module):
+    def __init__(self, tensor):
+        super().__init__()
+        self._tensor = tensor
+
+    def forward(self, original):
+        return self._tensor
+
+def _reparametrize_in_submodule(module, path, tensor):
```

Comment:
nit you could refactor this and the remove into a single function that gets the submodule. And then the main function just set or remove a parametrization from it.

Reply:
sure!